### PR TITLE
Feature/tekton plugin initial code 

### PR DESCRIPTION
**Pull request by lokeshrangineni**
[https://github.com/janus-idp/backstage-plugins/issues/128 - Cecking in the initial Tekton plugin code.

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -109,3 +109,10 @@ catalog:
     #   target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme-corp.yaml
     #   rules:
     #     - allow: [User, Group]
+
+tekton:
```
Comment (riginoommen) [inline app-config.yaml:113]: Utilize the env variables for managing configurations like secrets

File: plugins/tekton-pipelines-backend/src/service/pipelinerun.ts [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+  baseUrl: string,
+  authorizationBearerToken: string,
+  namespace: string,
+  selector: string,
+  dashboardBaseUrl: string,
+): Promise<PipelineRun[]> => {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+  
+  console.log("Logging the env variable", process.env.NODE_TLS_REJECT_UNAUTHORIZED);
+  //console.log("Logging the env variable", process.env);
```
Comment (riginoommen) [inline plugins/tekton-pipelines-backend/src/service/pipelinerun.ts:23]:
```suggestion
```

File: plugins/tekton-pipelines-backend/src/service/pipelinerun.ts [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    url = `${baseUrl}/apis/tekton.dev/v1beta1/namespaces/${namespace}/pipelineruns?labelSelector=${selector}`;
+  } else {
+    url = `${baseUrl}/apis/tekton.dev/v1beta1/namespaces/${namespace}/pipelineruns`;
+  }
+  let response: any
+  if (authorizationBearerToken !== "") {
+    response = await fetch(url, {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${authorizationBearerToken}`,
```
Comment (riginoommen) [inline plugins/tekton-pipelines-backend/src/service/pipelinerun.ts:35]: Append the ```Bearer``` keyword also in the env variable token

File: plugins/tekton-pipelines-backend/src/service/pipelinerun.ts [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+const getPipelineRuns = async (
+  baseUrl: string,
+  authorizationBearerToken: string,
+  namespace: string,
+  selector: string,
+  dashboardBaseUrl: string,
+): Promise<PipelineRun[]> => {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+  
+  console.log("Logging the env variable", process.env.NODE_TLS_REJECT_UNAUTHORIZED);
```
Comment (riginoommen) [inline plugins/tekton-pipelines-backend/src/service/pipelinerun.ts:22]:
```suggestion

```

File: plugins/tekton-pipelines-backend/src/service/router.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            stepContainer,
+          )
+
+          response.send(logs)
+        } 
+      }      
+    })    
+  
+  router.get('/health', (_, response) => {
+    logger.info('PONG!');
```
Comment (riginoommen) [inline plugins/tekton-pipelines-backend/src/service/router.ts:101]:
```suggestion
```

I think the APi response is sufficient

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: plugins/tekton-pipelines/src/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+export const TektonPipelinesPluginPage: () => JSX.Element = tektonPipelinesPluginPlugin.provide(
+  createRoutableExtension({
+    name: 'TektonPipelinesPluginPage',
+    component: () =>
+      import('./components/TektonDashboard').then(m => m.TektonDashboardComponent),
+    mountPoint: rootRouteRef,
+  }),
+);
+*/
+
```
Comment (riginoommen) [inline plugins/tekton-pipelines/src/plugin.ts:50]: unused comments should be removed

**Review by riginoommen — CHANGES_REQUESTED** [review]
I request to lint the code also

File: plugins/tekton-pipelines-backend/src/service/pipelinerun.ts [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+import fetch from 'node-fetch';
+
+const getPipelineRuns = async (
+  baseUrl: string,
+  authorizationBearerToken: string,
+  namespace: string,
+  selector: string,
+  dashboardBaseUrl: string,
+): Promise<PipelineRun[]> => {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
```
Comment (riginoommen) [inline plugins/tekton-pipelines-backend/src/service/pipelinerun.ts:20]: THis seems not a good idea

**Review by tumido — CHANGES_REQUESTED** [review]
Hey! Before we even attempt to bring this code in, please collaborate in the discussion on issue #128 to determine the right course of action here.

So far it seems we don't need to bring in anything else than the frontend tekton plugin.

Let's have a clear direction before we rush code in.

**Comment by lokeshrangineni** [comment]
Hi Tom,

Thank you for the review. I will incorporate your comments. How would you
prefer to have a conversation? Would you like to setup a call? Or I can
ping you as soon as i m back online.

Thanks,
Lokesh

**Comment by tumido** [comment]
Also if you can please come online on Slack for a moment, there's an issue that needs to be addressed asap.

**Comment by tumido** [comment]
Closing as a result of https://github.com/janus-idp/backstage-plugins/issues/161 discussion.

**Pull request closed without merging**